x-pack/solutions/observability/packages/kbn-streams-schema/src/helpers/type_guards.ts

```diff
@@ -47,6 +47,13 @@ export function isSchema<T>(zodSchema: ZodSchema, subject: T) {
   }
 }
 
+export function assertsSchema<TSchema extends ZodSchema>(
```

Comment:
this function throws if a schema doesn't match, instead of returning true/false, which means we get the original zod validation error